### PR TITLE
doc: add project history info page

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

File: doc/usr/expl/history.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+This project was initially developed in 2016 at a French startup company named
+Stupeflix. It took inspiration from the scene graph paradigm of [OSG] for which
+many clumsy in-house patches were maintained, essentially for video playback
+support and lazy resources management.
+
+The video playback logic was the first foundation to be extracted from these
+patches, and landed in a dedicated module named [sxplayer] (standing for
+"Stupeflix Player"), a free (LGPL) video playback library on top of FFmpeg.
+
+While it helps with maintenance, most of the OSG features remained unneeded,
```
Comment (mbouron) [inline doc/usr/expl/history.md:12]: helped ?

File: doc/usr/expl/history.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+the Python bindings were painful, and the complexity of the stack of patches
+added on top was awkward to manage. It was decided to prototype a new engine,
```
Comment (mbouron) [inline doc/usr/expl/history.md:14]: We should mention that the OSG version used at the time was frozen and was an already outdated version for the time. We were unable to rebase the whole stack of patches/hacks (including the custom python binding) back to upstream.

File: doc/usr/expl/history.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+and OpenGL, along with the fact that the scene graph is composed of "nodes".
+
+The project was open-sourced under Apache license when the [GoPro] company
+acquired Stupeflix (late 2016). The main authors of the project, Clément Bœsch
+and Matthieu Bouron, carried on with its development for years, making it a core
+element of the GoPro Quik mobile editing application.
+
+Along the years, the project matured into a significant engine for video
+compositing, animations, video playback, color management, and was notably
+entirely redesigned to support more modern graphics backends.
```
Comment (mbouron) [inline doc/usr/expl/history.md:29]: redesigned -> improved ? It's still a WIP (sorry).

File: doc/usr/expl/history.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Around the end of 2022 and beginning of 2023, Matthieu and Clément left the
+GoPro company and decided to maintain the project under a new name: the **Nope
+Project**. `sxplayer` and `node.gl` were respectively renamed to `nope.media`
+and `nope.gl`. The direction of the project then took a new turn, focusing on
+providing a free and open-source framework for motion design, 2D composition
+and visual effects.
+
+[OSG]: https://www.openscenegraph.com "Open Scene Graph"
+[sxplayer]: https://github.com/gopro/sxplayer "sxplayer"
+[GoPro]: https://gopro.com "GoPro"
```
Comment (mbouron) [inline doc/usr/expl/history.md:40]: Nice writing, thanks !

File: doc/usr/expl/history.md (hunk 1 above)
Comment (ubitux) [reply doc/usr/expl/history.md]: Oops fixed

File: doc/usr/expl/history.md (hunk 2 above)
Comment (ubitux) [reply doc/usr/expl/history.md]: Reworked that paragraph to add this information

File: doc/usr/expl/history.md (hunk 3 above)
Comment (ubitux) [reply doc/usr/expl/history.md:30]: I'm essentially referring to the fact that for years we've been extracting the GL bits out of the nodes and work on introducing new abstracts levels. I think that's accurate to say that we've entirely redesigned the project around this. This being 100% complete or not doesn't really matter.

File: doc/usr/expl/history.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Stupeflix. It took inspiration from the scene graph paradigm of [OSG] for which
+many clumsy in-house patches were maintained, essentially for video playback
+support and lazy resources management.
+
+The video playback logic was the first foundation to be extracted from these
+patches, and landed in a dedicated module named [sxplayer] (standing for
+"Stupeflix Player"), a free (LGPL) video playback library on top of FFmpeg.
+
+While it helped with maintenance, most of the OSG features remained unneeded,
+the Python bindings were unmaintained and painful to work with, the stack of
```
Comment (mbouron) [inline doc/usr/expl/history.md:13]: What do you think of:
```
the *custom* Python bindings were unmaintained, painful to work with and tied to a specific version of OSG, preventing us from using a more recent version. The stack of patches added on top of the project were also tied to the same specific version of OSG and were particularly challenging to rebase due to their complexity ?
```

Comment (ubitux) [reply doc/usr/expl/history.md]: Sure, replaced. I just replaced the "were" into "was", and "their" into "its" (*the* stack).

Unrelated but I also added "game engine" after the first mention of OSG in the first paragraph

**Review by mbouron — APPROVED** [review]

**Pull request merged**